src/py_avro_schema/_schemas.py

```diff
@@ -572,6 +573,8 @@ class UnionSchema(Schema):
     def handles_type(cls, py_type: Type) -> bool:
         """Whether this schema class can represent a given Python class"""
         origin = get_origin(py_type)
+        if getattr(types, "UnionType", None):
```

Comment:
I think a comment might be helpful to explain this conditional logic for Python 3.10+, including an explanation that `types.UnionType` is the `a | b` syntax.